runelite-client/src/main/java/net/runelite/client/plugins/timetracking/farming/FarmingContractManager.java

```diff
@@ -0,0 +1,322 @@
+		{
+			if (infoBox != null)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = null;
+			}
+		}
+		return oldSummary != summary;
+	}
+
+	private void handleInfoBox()
+	{
+		if (infoBox != null)
+		{
+			if (contract == null)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = null;
+			}
+			else if (infoBox.getContract() != contract)
+			{
+				infoBoxManager.removeInfoBox(infoBox);
+				infoBox = new FarmingContractInfoBox(itemManager.getImage(contract.getItemID()), plugin, contract, config, this);
+				infoBoxManager.addInfoBox(infoBox);
+			}
+		}
+		else if (contract != null)
+		{
+			infoBox = new FarmingContractInfoBox(itemManager.getImage(contract.getItemID()), plugin, contract, config, this);
+			infoBoxManager.addInfoBox(infoBox);
+		}
+	}
+
+	private void handleGuildmasterJaneWidgetDialog()
+	{
+		Widget npcDialog = client.getWidget(WidgetInfo.DIALOG_NPC_HEAD_MODEL);
+
+		if (npcDialog == null || npcDialog.getModelId() != GUILDMASTER_JANE_WIDGET_MODEL_ID)
+		{
+			return;
+		}
+
+		String dialogText = Text.removeTags(client.getWidget(WidgetInfo.DIALOG_NPC_TEXT).getText());
+
+		if (dialogText.equals(CONTRACT_REWARDED))
+		{
+			setContract(null);
+		}
+
+		Matcher matcher = CONTRACT_ASSIGN_PATTERN.matcher(dialogText);
+
+		if (!matcher.find())
+		{
+			return;
+		}
+
+		String name = matcher.group(1);
+
+		Produce farmingContract = Produce.getByContractName(name);
+
+		if (farmingContract == null)
+		{
+			return;
+		}
+
+		Produce currentFarmingContract = contract;
+
+		if (farmingContract == currentFarmingContract)
+		{
+			return;
+		}
+
+		setContract(farmingContract);
+	}
+
+	private void handleContractState()
+	{
+		if (contract == null)
+		{
+			summary = SummaryState.UNKNOWN;
+			return;
+		}
+
+		PatchImplementation patchImplementation = contract.getPatchImplementation();
+
+		boolean hasEmptyPatch = false;
+		for (FarmingPatch patch : farmingWorld.getFarmingGuildRegion().getPatches())
+		{
+			if (patch.getImplementation() != patchImplementation)
+			{
+				continue;
+			}
+
+			PatchPrediction prediction = farmingTracker.predictPatch(patch);
+
+			if (prediction == null)
+			{
+				continue;
+			}
+			Produce produce = prediction.getProduce();
+
+			if (produce == null || produce == Produce.WEEDS)
+			{
+				summary = SummaryState.EMPTY;
+				hasEmptyPatch = true;
+				continue;
+			}
+
+			if ((contract.requiresHealthCheck() && prediction.getCropState() == CropState.HARVESTABLE)
+				&& !hasEmptyPatch)
+			{
+				summary = SummaryState.OCCUPIED;
+			}
+
+			if (produce != contract)
+			{
+				if (!hasEmptyPatch)
+				{
+					summary = SummaryState.OCCUPIED;
+				}
+			}
+			else
+			{
+				long estimatedTime = prediction.getDoneEstimate();
+
+				if (estimatedTime <= Instant.now().getEpochSecond())
+				{
+					summary = SummaryState.COMPLETED;
+					completionTime = 0;
+				}
+				else
+				{
+					summary = SummaryState.IN_PROGRESS;
+					completionTime = estimatedTime;
+				}
+				break;
```

Comment:
If you have 2 patches that match this crop, and the first one is growing, but the second one is done, this will say you have to wait, which seems wrong.

Reply:
Tested this scenario (both ways) and some others after applying new fix.